### PR TITLE
Switch to poll SSH instead of API for inst ready

### PR DESCRIPTION
**Pull request by meatballhat**
by using the script upload retries to do the job

- [x] also rip out the legacy image lookup &lt;/shame&gt;
- [x] test in staging

### CODE REVIEW

**Comment by solarce** [comment]
A skim of it and it seems good to me but I'd want a review from @henrikhodne before merging, as my Go is pretty minimal

File: backend/gce_test.go [hunk 1]
```diff
@@ -176,6 +176,7 @@ func gceTestSetup(t *testing.T, cfg *config.ProviderConfig, resp *gceTestRespons
 		cfg = config.ProviderConfigFromMap(map[string]string{
 			"ACCOUNT_JSON": "{}",
 			"PROJECT_ID":   "project_id",
+			"IMAGE_ALIAS_": "default",
```
Comment (henrikhodne) [inline backend/gce_test.go]: This looks a little strange to me, what makes this necessary?

File: backend/gce.go [hunk 2]
```diff
@@ -539,62 +468,131 @@ func (p *gceProvider) Start(ctx gocontext.Context, startAttributes *StartAttribu
...
+	zoneOpCall := p.client.ZoneOperations.Get(p.projectID, p.ic.Zone.Name, op.Name)
+
+	for {
+		newOp, err := zoneOpCall.Do()
+		if err != nil {
+			errChan <- err
+			return multistep.ActionHalt
+		}
+
+		if newOp.Status == "RUNNING" {
```
Comment (henrikhodne) [inline backend/gce.go]: Is it possible that an operation for some reason goes from running to done without us noticing, so this branch never triggers? Maybe it should be `newOp.Status == "RUNNING" || newOp.Status == "DONE"`? I'm not super-familiar with the GCE API, so I may be misunderstanding things, though.

File: backend/gce_test.go (hunk 1 above)
Comment (meatballhat) [inline backend/gce_test.go]: It's totes strange, and specific to the tests, and means that language is `""` somewhere...

File: backend/gce.go (hunk 2 above)
Comment (meatballhat) [inline backend/gce.go]: Heck ya totes true

File: backend/gce_test.go (hunk 1 above)
Comment (meatballhat) [inline backend/gce_test.go]: found it

**Comment by henrikhodne** [comment]
Makes sense to me. How many "boot polls" is a normal boot? (Are we likely to run out of retries if boots get "a little" slower?)

**Comment by henrikhodne** [comment]
Other than that it looks good :sparkles: :+1: :heart:

**Comment by meatballhat** [comment]
@henrikhodne My logic goes kind a like: `(upload retries) * ((length of upload attempt) + (post-fail sleep)) > (instance start timeout)`, so I don't *think* it's an issue (??)

**Comment by henrikhodne** [comment]
> My logic goes kind a like: `(upload retries) * ((length of upload attempt) + (post-fail sleep)) > (instance start timeout)`, so I don't think it's an issue (??)

That looks right to me.

**Pull request merged**